lib/update-fetch.js

```diff
@@ -0,0 +1,201 @@
+var path = require('path');
+var urljoin = require('url-join');
+var request = require('request');
+var buildServerRoot = 'https://builds.tessel.io/t2';
+var firmwareBuildURL = urljoin(buildServerRoot, 'firmware');
+var buildsListURL = urljoin(firmwareBuildURL, 'builds.json');
+var openWRTBuild = 'openwrt.bin';
+var firmwareBuild = 'firmware.bin';
+var gunzip = require('zlib').createGunzip();
+var extract = require('tar-stream').extract();
+var logs = require('./logs');
+var streamToBuffer = require('stream-to-buffer');
+/*
+  Requests a list of available builds from the
+  build server. Returns list of build names in
+  a Promise.
+*/
+function fetchBuildList() {
+  return new Promise(function(resolve, reject) {
+    // Fetch the list of available builds
+    request.get(buildsListURL, function(err, response, body) {
+
+      // If there wasn't an issue with the request
+      if (requestRespHelper(resolve, reject, err, response)) {
+        // Resolve with the parsed data
+        try {
+          resolve(JSON.parse(body));
+        }
+        // If the parse failed, reject
+        catch (err) {
+          reject(err);
+        }
+      }
+    });
+  });
+}
+
+/* 
+  Accepts a build name and attempts to fetch
+  the build images from the server. Returns build contents
+  in a Promise
+*/
+function fetchBuild(version) {
+  return new Promise(function(resolve, reject) {
+    logs.info('Beginning update download. This could take a couple minutes..');
```

Comment:
"a couple of minutes" - also, how long should we expect? Is there something we can watch to make sure it's not just frozen?